lib/TPP/Transforms/TileConfig.cpp

```diff
@@ -0,0 +1,154 @@
+                                            xsmm::GemmFlags::NONE)) {
+      return;
+    }
+    attributes.push_back(flagItr);
+  }
+
+  if (attributes.empty())
+    attributes.push_back(
+        xsmm::GemmFlagsAttr::get(rewriter.getContext(), xsmm::GemmFlags::NONE));
+}
+
+template <typename InvokeOpTy, typename DispatchOpTy>
+struct TileConfig : OpRewritePattern<InvokeOpTy> {
+  using OpRewritePattern<InvokeOpTy>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(InvokeOpTy op,
+                                PatternRewriter &rewriter) const override {
+    if (xsmm::utils::getDataType(rewriter, op.getOperand(1).getType()) !=
+        xsmm::DataTypeAttr::get(rewriter.getContext(), xsmm::DataType::BF16))
+      return failure();
+    auto flags =
+        dyn_cast<DispatchOpTy>(op.getOperand(0).getDefiningOp()).getFlags();
+    for (auto flagItr : flags) {
+      if (flagItr == xsmm::GemmFlagsAttr::get(
+                         rewriter.getContext(),
+                         mlir::xsmm::GemmFlags::NO_RESET_TILECONFIG) ||
+          flagItr == xsmm::GemmFlagsAttr::get(
+                         rewriter.getContext(),
+                         mlir::xsmm::GemmFlags::NO_SETUP_TILECONFIG)) {
+        return failure();
+      }
+    }
+
+    SmallVector<Attribute> attributesSetup;
+    attributesSetup.push_back(xsmm::GemmFlagsAttr::get(
+        rewriter.getContext(), xsmm::GemmFlags::NO_RESET_TILECONFIG));
+    appendBrgemmFlags<InvokeOpTy, DispatchOpTy>(attributesSetup, rewriter, op);
+    auto tileConfigSetup = rewriter.create<xsmm::TileConfigDispatchOp>(
+        op.getLoc(), rewriter.getI64Type(),
+        DenseI64ArrayAttr::get(
+            rewriter.getContext(),
+            dyn_cast<DispatchOpTy>(op.getOperand(0).getDefiningOp())
+                .getInputs()),
+        rewriter.getArrayAttr(attributesSetup),
+        xsmm::utils::getDataType(rewriter, op.getOperand(1).getType()));
+
+    SmallVector<Attribute> attributesReset;
+    attributesReset.push_back(xsmm::GemmFlagsAttr::get(
+        rewriter.getContext(), xsmm::GemmFlags::NO_SETUP_TILECONFIG));
+    appendBrgemmFlags<InvokeOpTy, DispatchOpTy>(attributesReset, rewriter, op);
+    auto tileConfigReset = rewriter.create<xsmm::TileConfigDispatchOp>(
+        op.getLoc(), rewriter.getI64Type(),
+        DenseI64ArrayAttr::get(
+            rewriter.getContext(),
+            dyn_cast<DispatchOpTy>(op.getOperand(0).getDefiningOp())
+                .getInputs()),
+        rewriter.getArrayAttr(attributesReset),
+        xsmm::utils::getDataType(rewriter, op.getOperand(1).getType()));
+
+    SmallVector<Attribute> attributesBrgemm;
+    attributesBrgemm.push_back(xsmm::GemmFlagsAttr::get(
+        rewriter.getContext(), xsmm::GemmFlags::NO_RESET_TILECONFIG));
+    attributesBrgemm.push_back(xsmm::GemmFlagsAttr::get(
+        rewriter.getContext(), xsmm::GemmFlags::NO_SETUP_TILECONFIG));
+    appendBrgemmFlags<InvokeOpTy, DispatchOpTy>(attributesBrgemm, rewriter, op);
```

Comment:
I don't understand, you've just added (independently) the two flags above, why add the pair here?

Reply:
`appendBrgemmFlags` attributes are used in dispatch operation, whereas the other two attribute sets are used in tileconfigdispatch operations.

Reply:
@rengolin : this is just as we break API for performance, hence we need in total 3 ops and to not set it all time, it's negated.